dask/dataframe/core.py

```diff
@@ -7986,8 +8004,10 @@ def idxmaxmin_row(x, fn=None, skipna=True):
     minmax = "max" if fn == "idxmax" else "min"
     if len(x) > 0:
         x = x.set_index("idx")
-        idx = [getattr(x.value, fn)(skipna=skipna)]
-        value = [getattr(x.value, minmax)(skipna=skipna)]
+        # potentially coerced to object, so cast back
+        value = x.value.infer_objects()
```

Comment:
Interesting. Why didn't we need this one before?

Reply:
We didn’t support any non numerics before (bool is not supported), so we could never end up with object dtype